src/test/java/seedu/address/logic/parser/ClientNoteEditCommandParserTest.java

```diff
@@ -0,0 +1,90 @@
+package seedu.address.logic.parser;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NOTE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.ClientNoteEditCommand;
+import seedu.address.model.note.Note;
+import seedu.address.model.note.TagNoteMap;
+import seedu.address.model.tag.Tag;
+
+class ClientNoteEditCommandParserTest {
+    private static final String SPACE = " ";
+    private static final String CLIENT_INDEX_STRING = "1";
+    private static final Index CLIENT_INDEX = Index.fromOneBased(Integer.parseInt(CLIENT_INDEX_STRING));
+    private static final String NOTE_INDEX_STRING = "1";
+    private static final Index NOTE_INDEX = Index.fromOneBased(Integer.parseInt(NOTE_INDEX_STRING));
+    private static final String NOTE_STRING = "yay this be a note";
+    private static final String NO_CLIENT_INDEX_NO_NOTE_INDEX_NO_NOTE = "just an empty string here";
+    private static final String HAS_CLIENT_INDEX_HAS_NOTE_INDEX_NO_NOTE = CLIENT_INDEX_STRING + SPACE
+            + NOTE_INDEX_STRING + SPACE + " nope no note here";
+    private static final String HAS_CLIENT_INDEX_HAS_NOTE_INDEX_HAS_NOTE = CLIENT_INDEX_STRING + SPACE
+            + NOTE_INDEX_STRING + SPACE + PREFIX_NOTE + NOTE_STRING;
+    private static final String EXPECTED_PARSE_FAILURE_MESSAGE = String.format(MESSAGE_INVALID_COMMAND_FORMAT,
```

Comment:
I feel like these do not really help with making it easier to understand since it is pretty much repeated in the test names already. Possibly even makes it harder to understand since the user would have to match the name of the strings to find out what they test.

Reply:
added to issue #263